proto/sentry_protos/snuba/README.md

```diff
@@ -0,0 +1,48 @@
+# Disclaimer
+
+This project is not stable. At the moment it is being used to build the new performance product. Everything here is subject to change. DO NOT DEPEND ON THIS.
+
+# Qu-EAP
+
+This folder contains the interface definitions for querying of the events analytics platform: Qu-EAP. (Pronounced Kweep).
+
+
+# Terminology
+
+**Trace -** A collection of events that are triggered by a customer operation. Examples: A web request, an event going through a pipeline or all of those things grouped together (e.g. a web request triggers asynchronous processes which are still tied to the trace)
+
+**TraceItem -** A queryable structured attribute of a trace. Everything that is queryable through EAP is a **TraceItem or a trace.** TraceItems are related to each other via traces.
+**TraceItemAttribute -** An attribute on a traceitem that can be selected, aggregated (if it is numeric) or used to filter
+
+**Virtual Column -** Column that is not stored on the storage of the TraceItem but rather passed in at query time.
+
+Example:
+
+    `project_id` is stored with the trace_item
+    `project_name` is not
+    `project_id` to name mapping is passed with the query so that project_name can be a virtual column
+
+
+# Is this Protobuf definition a re-implementation of [SnQL](https://getsentry.github.io/snuba/language/snql.html)?
+
+No, Qu-EAP differes from SnQL in key ways:
+
+1. Qu-EAP does not distinguish between data stored as a column or a tag, TraceItems have attributes and the interface to query them is all the same.
+    1. Note: attributes sent by sentry are prefixed with `sentry.{attr_name}` to allow users to send custom tags with the same name.
```

Comment:
Should we do the reverse as well, prefix user tags with `user` or something similar? That way users can't send tags that overlap sentry tags by sending `sentry.blah`, it'll become `user.sentry.blah`

Reply:
that should presumably be blocked by sdks and in relay

Reply:
(it's against otel spec to use someone else's namespace)

Reply:
Good thing user's will read the entirety of otel spec then right ;)

Reply:
eh, ideally the SDKs can just tell them to knock it off

Reply:
🤔 what's the interface or experience you're imagining to tell them to knock it off?

Reply:
if you `span.setData("sentry.haha", "blah")` it says "you are not allowed to do that."

Reply:
🤔 via a warning? error? print statement?
does the data still get set? or do we null that data?
if `haha` is a key we rely on does `setData` still work? or becomes a no-op?

Reply:
Is there some reason not to namespace user data?